### PR TITLE
Flag examples in convolution sub-package

### PR DESCRIPTION
**Pull request by lglattly**
### Description

This pull request is to flag examples in the (previously copy edited) convolution sub-package for possible use in the example gallery.

cc @astropy/coordinators

### CODE REVIEW

**Comment by keflavich** [comment]
Looks fine at a glance; can I review this in final form somewhere?  These are example snippets that are going into the framework jonathansick showed, right?

**Comment by lglattly** [comment]
That's right! I'm working on putting in the markup on all of the examples so they can be reviewed for possible use. @jonathansick is there a way to preview these examples as they would look in the gallery or not at this stage?

File: docs/convolution/using.rst [hunk 1]
```diff
@@ -65,27 +69,37 @@ interpolated using the kernel::
     >>> convolve([1, 4, 5, 6, np.nan, 7, 8], [0.2, 0.6, 0.2], boundary='extend')  # doctest: +FLOAT_CMP
     array([1.6 , 3.6 , 5.  , 5.75, 6.5 , 7.25, 7.8 ])
 
+..
+  EXAMPLE END
+
+..
+  EXAMPLE START
+  Constructing Arrays as Lists
```
Comment (astrofrog) [inline docs/convolution/using.rst]: I think this should be Constructing Kernels from Lists

**Comment by jonathansick** [comment]
Sorry, there isn't a great way to preview these yet. Ultimately the comments will get converted into `example` directives, but we should wait on getting the Sphinx extension into the regular Astropy docs stack before doing that.

File: docs/convolution/using.rst (hunk 1 above)
Comment (lglattly) [reply docs/convolution/using.rst]: Thanks for the suggestion!

File: docs/convolution/index.rst [hunk 2]
```diff
@@ -153,7 +153,14 @@ algorithm, while :func:`~astropy.convolution.convolve_fft` uses a Fast Fourier
 Transform (FFT). Thus, the former is better for small kernels, while the latter
 is much more efficient for larger kernels.
 
-For example, to convolve a 1D dataset with a user-specified kernel, you can do::
+Example
+-------
+
+..
+  EXAMPLE START
+  Convolution for User-Specified Kernels
```
Comment (astrofrog) [inline docs/convolution/index.rst]: @jonathansick - just out of curiosity, is there any way we can make the markup more concise, e.g.

```
.. EXAMPLE START: Title

...

.. EXAMPLE END
```

?

Comment (jonathansick) [reply docs/convolution/index.rst]: That'd be fine. The comment syntax is just a human-readable/searchable placeholder for the actual syntax, which is

```rst
.. example:: title

   Example content.
```

Comment (lglattly) [reply docs/convolution/index.rst]: Good to know! So I should change the comment markup to:

.. EXAMPLE START: Title

Content

.. EXAMPLE END

with just one colon after "example start" correct?

Comment (jonathansick) [reply docs/convolution/index.rst]: Sounds good :) Starting by next week at the latest I'll start pushing again to get the extension shipped so you can use the real syntax at some point.

**Pull request merged**